### PR TITLE
Allow endpoint to end with '/'

### PR DESCRIPTION
**Pull request by zionts**
* Avoids copy/paste code for regex matching in express / connect
* Allows regex to match the endpoint ending with '/'
* Cleans up copy/paste code for '/graphql' endpoint in connect test

### CODE REVIEW

File: src/middleware.ts [hunk 1]
```diff
@@ -12,6 +12,12 @@ export class MiddlewareParams {
     public dumpTraffic: boolean;
 }
 
+// Matches the strict endpoint, the endpoint followed by one forward-slash, the endpoint
+// followed by a question mark and then anything else, and the endpoint followed by one backslash
+function getEndpointRegex(endpoint: string) : RegExp {
```
Comment (thepwagner) [inline src/middleware.ts]: Unit tests against this returned regexp might be sane - to demonstrate the addition and exercise edge cases (`/graphql/?foo=bar` isn't matched, maybe it should be?)

Comment (zionts) [reply src/middleware.ts]: That's a good point 👍  I'll add/edit before the merge

**Comment by glasser** [comment]
Interesting note: this fix appears to have only worked in double proxy mode! To make this test pass in single proxy mode, we need a Go improvement too.
